### ISSUE TITLE
componentWillReceiveProps warning with React 16.9

### ISSUE THREAD

**Issue by alexbensimon** [chenglou/react-motion#569]
Hello,

After updating `react` and `react-dom` to v`16.9.0`, this warning appears: 

```
react-dom.development.js:11494 Warning: componentWillReceiveProps has been renamed, and is not recommended for use. See https://fb.me/react-async-component-lifecycle-hooks for details.

* Move data fetching code or side effects to componentDidUpdate.
* If you're updating state whenever props change, refactor your code to use memoization techniques or move it to static getDerivedStateFromProps. Learn more at: https://fb.me/react-derived-state
* Rename componentWillReceiveProps to UNSAFE_componentWillReceiveProps to suppress this warning in non-strict mode. In React 17.x, only the UNSAFE_ name will work. To rename all deprecated lifecycles to their new names, you can run `npx react-codemod rename-unsafe-lifecycles` in your project source folder.

Please update the following components: Motion
```

According to https://reactjs.org/blog/2019/08/08/react-v16.9.0.html#renaming-unsafe-lifecycle-methods, the `componentWillReceiveProps` is now deprecated and will be removed from the API on the next major version.

**Comment by juicylevel** [comment]
2020 year..

**Comment by nkbt** [comment]
Yeah @juicylevel, I wish I could drop all the things of my life now (as in work, surfing, divorce, girlfriend) and do all the opensource...

On a side note: I've added a reminder to publish new version of react-motion which should have this fixed (at least temporarily). I'll let everyone know when it is published. You can help by teting current master branch in your real-world project and see if anything breaks.

**Comment by jfroffice** [comment]
Any update on this ?

**Comment by comp615** [comment]
When is the reminder to publish set for? Are there any other maintainers who might be able to help you out and publish this in the meantime? I know it's annoying to have people nagging, but getting the newer React support updated would be super helpful! Thank you!

**Comment by fabiomcosta** [comment]
Ah interesting... so sounds like the change is already there and we just need a new version published to `npm`?
If the maintainers give me the permissions I'll happily do that.

**Comment by andresmatasuarez** [comment]
yes, please!

**Comment by LiteSoul** [comment]
...and now it's 2021 🥺

**Comment by RaddishIoW** [comment]
Someone has forked react-motion to fix this issue: [see this thread](https://github.com/chenglou/react-motion/issues/604)

**Comment by onionhammer** [comment]
Perhaps other projects should reference the fork

**Comment by RaddishIoW** [comment]
@nkbt Any update on this? If there's a need for help with maintaining this, I'm sure there's competent people here willing to help...

**Comment by nkbt** [comment]
I am still getting notifications and emails, btw =)
It is not just the matter of approving PR and pushing new release unfortunately (oh I wish it would be)

Here is some context. It is hard for me to test react-motion in real-world product as I no longer work with any projects using react-motion. So the only way for me now would be to use demo examples only (I remember having a few edge cases beyond those). Also I never really used or worked with anything except main Motion component (plus core), so have basically no idea about things like staggered motion, etc. There was partially completed refactoring which we did not get finished as contributor vanished and I did not have any spare time to finalise all the work myself, thus releasing new version to the public was blocked (still is, as it turns out). Should have reverted all back in hindsight...
@RaddishIoW There were few cases people jumping on and helping out with the library one or a few times and then leaving, so not like there was anyone specific committed to do full on support work on it.
Generally it's quite hard to squeeze opensource in between all the life things like family, baby, travelling, work, personal projects, unless it is used in your work projects and you have ability to contribute as part of your day job (which is not the case for me anymore for quite a while now).
I'll see if I can do anything, but cannot promise much at this point. I want to make sure it is in shape to safely do patch releases without breaking changes (check past PRs and revert unfinished work from master). Then it would be in a maintenance mode (just keep updating critical deps, etc).

**Comment by Me-UserName** [comment]
2021

**Comment by JohnRoseDev** [comment]
How can an amazing project with 20k stars, conf presence (react europe talk), and a fantastic api be abandoned like this by the owner(s)? Completely understand if they don't have time anymore for the project, no blame in that, but couldn't one try in that case to delegate some ownership so someone can simply publish a new version of this package since the necessary code change is already there?

**Comment by catamphetamine** [comment]
2022

**Comment by theformer** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by GPrimola** [comment]
2023

**Comment by theformer** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by datplt** [comment]
2024

**Comment by theformer** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。

**Comment by am2619015** [comment]
Here we are, almost in 2025, and still nothing. This is abandoned.

**Comment by theformer** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
